src/http/content_encoding.rs

```diff
@@ -122,30 +128,28 @@ pub fn get_prior_encoding<'a>(accept_encoding: &'a HeaderValue) -> &'static str
         .unwrap_or(IDENTITY)
 }
 
-/// Compress data.
+/// Compress data stream.
 ///
 /// # Parameters
 ///
-/// * `data` - Data to be compressed.
-/// * `encoding` - Only support `br`, `gzip`, `deflate` and `identity`.
-pub fn compress(data: &[u8], encoding: &str) -> io::Result<Vec<u8>> {
-    use std::io::prelude::*;
-    let mut buf = Vec::new();
+/// * `input` - [`futures::stream::Stream`](futures::stream::Stream) to be compressed e.g. [`hyper::body::Body`](hyper::body::Body).
```

Comment:
One trick is that 

```
[`YourType`]
```

is effective to generating an intra doc link. [^1]

Also, A shorter type looks nice, at least the readability of the source is better.


```suggestion
/// * `input` - [`futures::Stream`] to be compressed, e.g. [`hyper::Body`].
```

[^1]: https://rust-lang.github.io/rfcs/1946-intra-rustdoc-links.html#implied-shortcut-reference-links

Reply:
Applied on 2550654